src/components/task-plan/plan-mixin.coffee

```diff
@@ -57,9 +58,23 @@ module.exports =
     courseId = @props.courseId
     TaskPlanActions.saved.removeListener('change', @saved)
     TaskPlanStore.isLoading(@props.id)
-    @context.router.goBack()
+    @goBackToCalendar()
 
   cancel: ->
     {id} = @props
-    TaskPlanActions.reset(id)
-    @context.router.goBack()
+    if confirm('Are you sure you want to cancel?')
+      TaskPlanActions.resetPlan(id)
+      @goBackToCalendar()
+
+  goBackToCalendar: ->
+    {id, courseId} = @props
+    calendarRoute = 'calendarByDate'
+    dueAt = TaskPlanStore.getFirstDueDate(id) or @context?.router?.getCurrentQuery()?.due_at
+    date = moment(dueAt).format('YYYY-MM-DD')
```

Comment:
if `@context.router.getCurrentQuery().due_at` doesn't exist, it's okay since moment will create a date for today...actually lemme push in a thing that'll use TimeStore.getNow() in that case to be better